{
  "uid" : "441fdc13dc8b98bb",
  "name" : "POST Создать пользователя с корректными данными",
  "fullName" : "tests.test_api.test_api.TestAPI#test_post_auth_user",
  "historyId" : "d9a2437627bbe237bdf3b37bffffa3b2",
  "time" : {
    "start" : 1714097882029,
    "stop" : 1714097893153,
    "duration" : 11124
  },
  "status" : "skipped",
  "statusMessage" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
  "statusTrace" : "self = <tests.test_api.test_api.TestAPI object at 0x7f50873be2a0>\n\n    @pytest.mark.xfail\n    @allure.title(\"POST Создать пользователя с корректными данными\")\n    def test_post_auth_user(self):\n        url = f'{self.constant.BASE_URL}auth/users/'\n        response = requests.post(url, json=self.constant.CREATE_USER)\n        time.sleep(10)\n>       Assertions.assert_status_code(response, self.code.STATUS_201)\n\ntests/test_api/test_api.py:27: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nresponse = <Response [400]>, expected_status_code = 201\n\n    @staticmethod\n    def assert_status_code(response: Response, expected_status_code: int):\n        actual_status_code = response.status_code\n        with allure.step(f\"Expected status {expected_status_code}\"):\n>           assert actual_status_code == expected_status_code, \\\n                f\"Unexpected status code. Expected: {expected_status_code}. Actual: {actual_status_code}\"\nE           AssertionError: Unexpected status code. Expected: 201. Actual: 400\n\napi_testing/assertions.py:11: AssertionError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1714097882013,
      "stop" : 1714097882029,
      "duration" : 16
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "clear_allure_results_folder",
    "time" : {
      "start" : 1714097882029,
      "stop" : 1714097882029,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "status" : "skipped",
    "statusMessage" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
    "statusTrace" : "self = <tests.test_api.test_api.TestAPI object at 0x7f50873be2a0>\n\n    @pytest.mark.xfail\n    @allure.title(\"POST Создать пользователя с корректными данными\")\n    def test_post_auth_user(self):\n        url = f'{self.constant.BASE_URL}auth/users/'\n        response = requests.post(url, json=self.constant.CREATE_USER)\n        time.sleep(10)\n>       Assertions.assert_status_code(response, self.code.STATUS_201)\n\ntests/test_api/test_api.py:27: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nresponse = <Response [400]>, expected_status_code = 201\n\n    @staticmethod\n    def assert_status_code(response: Response, expected_status_code: int):\n        actual_status_code = response.status_code\n        with allure.step(f\"Expected status {expected_status_code}\"):\n>           assert actual_status_code == expected_status_code, \\\n                f\"Unexpected status code. Expected: {expected_status_code}. Actual: {actual_status_code}\"\nE           AssertionError: Unexpected status code. Expected: 201. Actual: 400\n\napi_testing/assertions.py:11: AssertionError",
    "steps" : [ {
      "name" : "Expected status 201",
      "time" : {
        "start" : 1714097893152,
        "stop" : 1714097893152,
        "duration" : 0
      },
      "status" : "failed",
      "statusMessage" : "AssertionError: Unexpected status code. Expected: 201. Actual: 400\n",
      "statusTrace" : "  File \"/home/runner/work/PulseWave/PulseWave/api_testing/assertions.py\", line 11, in assert_status_code\n    assert actual_status_code == expected_status_code, \\\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 1,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Тестирование API"
  }, {
    "name" : "tag",
    "value" : "@pytest.mark.xfail"
  }, {
    "name" : "parentSuite",
    "value" : "tests.test_api"
  }, {
    "name" : "suite",
    "value" : "test_api"
  }, {
    "name" : "subSuite",
    "value" : "TestAPI"
  }, {
    "name" : "host",
    "value" : "fv-az1240-635"
  }, {
    "name" : "thread",
    "value" : "2295-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_api.test_api"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 0,
        "skipped" : 9,
        "passed" : 0,
        "unknown" : 0,
        "total" : 9
      },
      "items" : [ {
        "uid" : "cf33291fc4561bc0",
        "reportUrl" : "https://Nat754.github.io/PulseWave/5//#testresult/cf33291fc4561bc0",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1714013808007,
          "stop" : 1714013819007,
          "duration" : 11000
        }
      }, {
        "uid" : "7ebb2344eb9a31be",
        "reportUrl" : "https://Nat754.github.io/PulseWave/211//#testresult/7ebb2344eb9a31be",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1713945247871,
          "stop" : 1713945258821,
          "duration" : 10950
        }
      }, {
        "uid" : "f672fef5ff2be685",
        "reportUrl" : "https://Nat754.github.io/PulseWave/4//#testresult/f672fef5ff2be685",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1713925073848,
          "stop" : 1713925084793,
          "duration" : 10945
        }
      }, {
        "uid" : "cf2ffbdf0358eba7",
        "reportUrl" : "https://Nat754.github.io/PulseWave/211//#testresult/cf2ffbdf0358eba7",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1713900878471,
          "stop" : 1713900889550,
          "duration" : 11079
        }
      }, {
        "uid" : "30ea5ec83cc4d9cb",
        "reportUrl" : "https://Nat754.github.io/PulseWave/210//#testresult/30ea5ec83cc4d9cb",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1713884974879,
          "stop" : 1713884986102,
          "duration" : 11223
        }
      }, {
        "uid" : "4119f92fb3141989",
        "reportUrl" : "https://Nat754.github.io/PulseWave/209//#testresult/4119f92fb3141989",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1713882917336,
          "stop" : 1713882928595,
          "duration" : 11259
        }
      }, {
        "uid" : "c7c6d336566b3fba",
        "reportUrl" : "https://Nat754.github.io/PulseWave/209//#testresult/c7c6d336566b3fba",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1713881392683,
          "stop" : 1713881404067,
          "duration" : 11384
        }
      }, {
        "uid" : "c88f0e2a473216a9",
        "reportUrl" : "https://Nat754.github.io/PulseWave/208//#testresult/c88f0e2a473216a9",
        "status" : "skipped",
        "statusDetails" : "XFAIL\n\nAssertionError: Unexpected status code. Expected: 201. Actual: 400",
        "time" : {
          "start" : 1713878570412,
          "stop" : 1713878582451,
          "duration" : 12039
        }
      } ]
    },
    "tags" : [ "@pytest.mark.xfail" ]
  },
  "source" : "441fdc13dc8b98bb.json",
  "parameterValues" : [ ]
}